light_node/src/main.rs

```diff
@@ -242,7 +245,63 @@ fn block_on_actors(
     });
 }
 
+unsafe fn enable_core_dumps() {
+    let mut rlim = libc::rlimit{rlim_cur: 0, rlim_max: 0};
+    // Get current size limit of core dumps
+    if libc::getrlimit(libc::RLIMIT_CORE, &mut rlim) != 0 {
+        let err = io::Error::last_os_error();
+        println!("Enabling core dumps failed (getrlimit): {}", err);
+        return;
+    }
+
+    // Bump soft limit to the hard limit, but only if it was not set before
+    if rlim.rlim_cur == 0 {
+        rlim.rlim_cur = rlim.rlim_max;
+        if rlim.rlim_cur > 0 {
+            if libc::setrlimit(libc::RLIMIT_CORE, &rlim) != 0 {
+                let err = io::Error::last_os_error();
+                println!("Enabling core dumps failed (setrlimit): {}", err);
+                return;
+            }
+            println!("Core dumps enabled with maximum size.");
+        }
+    }
+}
+
+unsafe fn set_file_desc_limit(num : u64) {
+    let mut rlim = libc::rlimit{rlim_cur: 0, rlim_max: 0};
+    // Get current open file desc limit
+    if libc::getrlimit(libc::RLIMIT_NOFILE, &mut rlim) != 0 {
+        let err = io::Error::last_os_error();
+        println!("Setting open files limit failed (getrlimit): {}", err);
+        return;
+    }
+
+    // If the currently set rlimit is higher, do not change it
+    if rlim.rlim_cur >= num {
+        return;
+    }
+    // Set rlimit to num, but not higher than hard limit
+    rlim.rlim_cur = cmp::min(num, rlim.rlim_max);
+    if libc::setrlimit(libc::RLIMIT_NOFILE, &rlim) != 0 {
+        let err = io::Error::last_os_error();
+        println!("Setting open files limit failed (setrlimit): {}", err);
+        return;
+    }
+    println!("Open files limit set to {}.", rlim.rlim_cur);
+}
+
 fn main() {
+    // Need unsafe to access direct libc syscall interface
```

Comment:
@michal-stanek 
and also here, just call something like:
```
system::init_limits();
```

or maybe:
```
system::init_limits(log, env);
```
and that constant **64 * 1024** could be put into the configuration as optional, just to let users possibility to set it according to their needs and default behavior will be your algo here

now, the constant is here: `env.storage.db_cfg.max_open_files` in `DbConfiguration`


is it ok set thouse rlimits after creating logger and env? or this rlimits stuff have to be first step in main.rs?
what do you thing?

Reply:
ok, ok, makes sense,
so pls. remove `env.storage.db_cfg.max_open_files`

Reply:
Updated the PR. I ended up using the rlimit crate instead of libc - it does look a bit nicer. I also moved the initialization after log and env as you suggested - it shouldn't make a difference and we can use logger instead of print.
I confirmed that protocol runner processes inherit the rlimits.
Please check if it looks okay, then I'll squash the commits. I'll also add a separate commit removing the db_cfg.max_open_files.

Reply:
@michal-stanek 
ok, cool, looks much nicer,

please, remove that db_cfg.max_open_files, if we dont need to set it.
or what if we set db_cfg.max_open_files as resolved rlimit from set_file_desc_limit?
that we could avoid rocksdb too_many_open_files error?

Reply:
I'd remove it altogether. If we set max_open_files to -1 (recommended setting with respect to RocksDB performance) this means unlimited, but I wouldn't expect RocksDB to ever get any close to 64k open files. The RocksDB stats monitoring which we will have in the future will also include checks for how many files are open. I also expect that the other bounds which we will be setting in RocksDB (LSM file sizes, etc) will indirectly limit the number of files open to some 'small' amount - like not more than a few hundred. We will have tight control over those values hopefully :)

This also depends on whether we can get the other JIRA issue resolved (panic and crash due to too many open files). RocksDB should be handling this error correctly and adjusting (should consolidate files then), but the crash we experienced is probably due to some other reason - I guess it's not handled properly in some other place. But if it turns out that we cannot find where it happens and fix it, then we can limit db_max_open_files to <64k as a workaround.

Also bear in mind the rlimit is for all file descriptors so that also includes sockets, etc.

Reply:
I did some more reading (web + internal RocksDB code) and now I'm even more convinced we should keep the default RocksDB setting for max_open_files. If in any situation we would want to have a small number of files generated by RocksDB, we can adjust size multipliers for SST files.
I updated the PR.

Reply:
ok, cool, thank you